src/Bicep.Core/Semantics/SemanticModelHelper.cs

```diff
@@ -71,7 +71,7 @@ public static IEnumerable<FunctionCallSyntaxBase> GetFunctionsByName(SemanticMod
             // when we inevitably add a third language ID,
             // the inclusion list style below will prevent the new language ID from being
             // automatically allowed to be referenced via module declarations
-            if (sourceFile is not (BicepFile or ArmTemplateFile or TemplateSpecFile))
+            if ((reference is ExtendsDeclarationSyntax && sourceFile is not BicepParamFile) || (reference is not ExtendsDeclarationSyntax && sourceFile is not (BicepFile or ArmTemplateFile or TemplateSpecFile)))
```

Comment:
This logic is getting more complex - how about refactoring to something like the following:
```csharp
var isValidReference = reference switch {
    ExtendsDeclarationSyntax => sourceFile is BicepParamFile,
    _ => sourceFile is BicepFile or ArmTemplateFile or TemplateSpecFile,
};

if (!isValidReference)
...
```

You can get more fancy with pattern matching if you like:
```csharp
var isValidReference = (reference, sourceFile) switch {
    (ExtendsDeclarationSyntax, BicepParamFile) => true,
    (_, BicepFile or ArmTemplateFile or TemplateSpecFile) => true,
    _ => false,
};
```

Reply:
love the suggestion, implemented it 👍